core-processor/src/executor.rs

```diff
@@ -164,7 +164,16 @@ pub fn execute_wasm<A: ProcessorExt + EnvExt + IntoExtInfo + 'static, E: Environ
         Default::default(),
     );
 
-    let lazy_pages_enabled = ext.try_to_enable_lazy_pages(program_id, initial_pages);
+    let lazy_pages_enabled = match ext.try_to_enable_lazy_pages(program_id, initial_pages) {
+        Ok(enabled) => enabled,
+        Err(e) => {
+            return Err(ExecutionError {
+                program_id,
+                gas_amount: ext.into_gas_amount(),
+                reason: e,
+            })
+        }
+    };
```

Comment:
```suggestion
    let lazy_pages_enabled = ext.try_to_enable_lazy_pages(program_id, initial_pages).map_err(|reason| ExecutionError::from(program_id, gas_amount, reason)?;
```
So `from` or `new` for `ExecutionError` must be implemented. What do you think?

Reply:
What is gas_amount variable? How you wanna to initialise it?